lib/Sema/CSApply.cpp

```diff
@@ -7497,11 +7497,18 @@ static bool isPotentialPropertyWrapperInit(Expr *baseExpr,
   return true;
 }
 
+/// Detect whether the given baseExpr is a borrow.
+static bool isBorrowExprOfLValueType(Expr *baseExpr) {
+  if (auto *pd = dyn_cast<ParenExpr>(baseExpr))
+    baseExpr = pd->getSubExpr();
```

Comment:
make sure this handles arbitrary nestings like `((borrow x))`; can just make it recursive.

Reply:
We should avoid using recursion in the compiler. Often times if the compiler runs on side threads we have much less stack space than one may think. Better to just use a while loop.